src/main/java/com/synopsys/integration/detect/lifecycle/boot/product/ProductBoot.java

```diff
@@ -92,13 +92,26 @@ private BlackDuckRunData getBlackDuckRunData(
         if (blackDuckConnectivityResult.isSuccessfullyConnected()) {
             BlackDuckServicesFactory blackDuckServicesFactory = blackDuckConnectivityResult.getBlackDuckServicesFactory();
 
+            BlackDuckRunData bdRunData= null;
             if (shouldUsePhoneHome(analyticsConfigurationService, blackDuckServicesFactory.getApiDiscovery(), blackDuckServicesFactory.getBlackDuckApiClient())) {
                 PhoneHomeManager phoneHomeManager = productBootFactory.createPhoneHomeManager(blackDuckServicesFactory);
-                return BlackDuckRunData.online(blackDuckDecision.scanMode(), blackDuckServicesFactory, phoneHomeManager, blackDuckConnectivityResult.getBlackDuckServerConfig());
+                bdRunData = BlackDuckRunData.online(blackDuckDecision.scanMode(), blackDuckServicesFactory, phoneHomeManager, blackDuckConnectivityResult.getBlackDuckServerConfig());
             } else {
                 logger.debug("Skipping phone home due to Black Duck global settings.");
-                return BlackDuckRunData.onlineNoPhoneHome(blackDuckDecision.scanMode(), blackDuckServicesFactory, blackDuckConnectivityResult.getBlackDuckServerConfig());
+                bdRunData = BlackDuckRunData.onlineNoPhoneHome(blackDuckDecision.scanMode(), blackDuckServicesFactory, blackDuckConnectivityResult.getBlackDuckServerConfig());
             }
+            if (bdRunData.isRapid() && blackDuckDecision.hasSignatureScan()) {
+                String minBlackDuckVersionForRapidSignatureScan = "2022.10.0";
+                String bdVersion = blackDuckConnectivityResult.getContactedServerVersion();
+                if (minBlackDuckVersionForRapidSignatureScan.compareTo(bdVersion) > 0) {
+                    // abort!
+                    throw new DetectUserFriendlyException(
+                            "Cannot use RAPID SIGNATURE SCAN with Black Duck Versions prior to 2022.10.0!  The Black Duck Version attempted was: " + bdVersion,
+                            ExitCodeType.FAILURE_BLACKDUCK_CONNECTIVITY
+                        );
+                }
```

Comment:
It pains me that we have started down this slippery slope. But since we have, I think we'll want a mechanism to deal with it. This does seem like the right place to call it, but I think it should be it's own class (an injected dependency) that knows the minimum version for different features (today it would just know about this one feature). I'm thinking we'll need an enum of features, and a map that has feature -> min version).

Also, string compare won't work for this, will it? String compare would decide that 2022.2.0 is later than 2022.10.0, right?

Reply:
that is a good point and I admit I dont have a good answer for it.  what can I expect the form of the version to be?  Is this form guaranteed?

Reply:
Not sure what to say about this however I think I have a better handle on the version compare.